File: 프로그래머스/lv2/42842. 카펫/카펫.py
# 8 <= 갈색 <= 5,000
# 1 <= 노랑 <= 2,000,000
# 가로 >= 세로

# def solution(brown, yellow):    
#     for L in range(1, yellow // 2 + 1):
        
#         if yellow % L == 0: 
#             W = yellow // L 
#         else : continue
        
#         if (W * 2) + (L * 2) + 4 == brown: 
#             return [W + 2,L + 2]
        
def solution(brown, yellow):
    
    for i in range(1,yellow//2+2):
        if yellow % i == 0:
            length = i# 세로
            width = yellow // i # 가로
            
        if (length*2)+(width*2)+4 == brown:
            return [width+2,length+2]
    

File: 프로그래머스/2/42842. 카펫/카펫.js
function solution(brown, yellow) {
    const cnt = brown + yellow
    let height = 3, length = 0
    
    while(true) {
        if (cnt % height === 0) {
            length = cnt / height
            if ( (height-2) * (length-2) === yellow){
                break
            }
        }
        height++
    }
    
    return [length, height];
}